rustsec/src/cached_index.rs

```diff
@@ -6,6 +6,28 @@ use crate::{
     package::{self, Package},
 };
 
+pub use tame_index::external::reqwest::blocking::ClientBuilder;
+
+enum Index {
+    Git(tame_index::index::RemoteGitIndex),
+    SparseCached(tame_index::index::SparseIndex),
+    SparseRemote(tame_index::index::RemoteSparseIndex),
+}
+
+impl Index {
+    #[inline]
+    fn krate(&self, name: &package::Name) -> Result<Option<tame_index::IndexKrate>, Error> {
```

Comment:
To make sparse index fetch not terribly slow, this function will have to be removed. Instead need an API that accepts a bunch of package names all at once, and returns a map of package names to package metadata.